src/cryptenroll/cryptenroll-tpm2.c

```diff
@@ -160,6 +163,21 @@ int enroll_tpm2(struct crypt_device *cd,
                 r = get_pin(&pin_str, &flags);
                 if (r < 0)
                         return r;
+
+                r = crypto_random_bytes(binary_salt, sizeof(binary_salt));
+                if (r != 0)
+                        return 1;
+
+                uint8_t salted_pin[SHA256_DIGEST_SIZE] = {};
+                tpm2_util_pbkdf(pin_str, strlen(pin_str), binary_salt, sizeof(binary_salt), 1000, salted_pin);
+
+                explicit_bzero_safe(pin_str, strlen(pin_str));
+                mfree(pin_str);
+                pin_str = NULL;
+                /* re-stringify pin_str */
+                r = base64mem(salted_pin, sizeof(salted_pin), &pin_str);
```

Comment:
Question from @williamcroberts : r is in int but base64mem returns ssize_t, this was copied from enroll_fido2 from line 56.

Reply:
ah, fuck. this is borked all over the place actually, will look into this. thanks for the pointer.

Reply:
→ https://github.com/systemd/systemd/pull/26016